server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -381,6 +382,114 @@ func (s *BuildBuddyServer) JoinGroup(ctx context.Context, req *grpb.JoinGroupReq
 	return &grpb.JoinGroupResponse{}, nil
 }
 
+func (s *BuildBuddyServer) GetApiKeys(ctx context.Context, req *akpb.GetApiKeysRequest) (*akpb.GetApiKeysResponse, error) {
+	userDB := s.env.GetUserDB()
+	if userDB == nil {
+		return nil, status.UnimplementedError("Not Implemented")
+	}
+	groupID := req.GetGroupId()
+	if err := s.authorizeGroupAccess(ctx, groupID); err != nil {
+		return nil, err
+	}
+	tableKeys, err := userDB.GetAPIKeys(ctx, groupID)
+	if err != nil {
+		return nil, err
+	}
+	protoKeys := make([]*akpb.ApiKey, len(tableKeys))
+	for i, k := range tableKeys {
+		protoKeys[i] = &akpb.ApiKey{
+			Id:    k.APIKeyID,
+			Value: k.Value,
+			Label: k.Label,
+		}
+	}
+	return &akpb.GetApiKeysResponse{
+		ApiKey: protoKeys,
+	}, nil
+}
+
+func (s *BuildBuddyServer) CreateApiKey(ctx context.Context, req *akpb.CreateApiKeyRequest) (*akpb.CreateApiKeyResponse, error) {
+	userDB := s.env.GetUserDB()
+	if userDB == nil {
+		return nil, status.UnimplementedError("Not Implemented")
+	}
+	groupID := req.GetGroupId()
+	if err := s.authorizeGroupAccess(ctx, groupID); err != nil {
+		return nil, err
+	}
+	k, err := userDB.CreateAPIKey(ctx, groupID, req.GetLabel())
+	if err != nil {
+		return nil, err
+	}
+	return &akpb.CreateApiKeyResponse{
+		ApiKey: &akpb.ApiKey{
+			Id:    k.APIKeyID,
+			Value: k.Value,
+			Label: k.Label,
+		},
+	}, nil
+}
+
+func (s *BuildBuddyServer) authorizeAPIKeyAccess(ctx context.Context, apiKeyID string) error {
```

Comment:
can you replace this with a standard permissions check?

Reply:
Done